### PR TITLE
Maryia/DTRA-403/feat: migrate Routes files in Trader package to TS

### PR DESCRIPTION
**Pull request by maryia-deriv**
## Changes:

- to migrate Routes files in Trader package to TS

### Screenshots:

Please provide some screenshots of the change.

### CODE REVIEW

File: packages/components/src/components/dialog/dialog.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ type TDialog = {
     onConfirm: () => void;
     onEscapeButtonCancel?: () => void;
     portal_element_id?: string;
-    title?: string;
+    title?: React.ReactNode;
```
Comment (maryia-deriv) [inline packages/components/src/components/dialog/dialog.tsx:29]: This helps to resolve type issue when common-store error is passed to ErrorComponent and further to this Dialog. Also, this title is rendered directly in jsx, so I believe that this change will facilitate future replacement of localizes, which are passed as titles to Dialog, with Localize components.

**Comment by maryia-deriv** [comment]
all tests are passing locally

File: packages/trader/src/App/Containers/Routes/routes.tsx [hunk 2]
```diff
@@ -9,11 +8,29 @@ import getRoutesConfig from 'App/Constants/routes-config';
 import { observer, useStore } from '@deriv/stores';
 import { useTraderStore } from 'Stores/useTraderStores';
 
-const checkRoutingMatch = (route_list, path) => {
+type TMatchPattern = { from: Array<string | undefined>; to: string[] };
```
Comment (kate-deriv) [inline packages/trader/src/App/Containers/Routes/routes.tsx:11]: Maybe it's better to stick to one code style: Array<some_type> or some_type[]?

Comment (maryia-deriv) [reply packages/trader/src/App/Containers/Routes/routes.tsx:11]: @kate-deriv at least on one line, yes, i'll update this, thank you  :)

**Review by kate-deriv — APPROVED** [review]

**Review by kate-deriv — APPROVED** [review]

**Review by henry-deriv — APPROVED** [review]

**Review by akmal-deriv — APPROVED** [review]

**Pull request merged**